gen/calc.py

```diff
@@ -575,6 +575,30 @@ def fz_err(msg):
             validate_int_in_range(port, 1, 65535)
 
 
+def calculate_scarce_resources(enable_scarce_resources):
```

Comment:
I talked with Maksym about this and its clear what he is doing here. It's not validation, but rather a calculation of whether this flag should be set to true or false based on its contents.